elements/lisk-tree/src/merkle_tree/types.ts

```diff
@@ -37,6 +38,12 @@ export const enum NodeSide {
 export interface TreeStructure {
 	[key: number]: NodeInfo[];
 }
+
+export interface NodeIndex {
+	layerIndex: number | undefined;
+	nodeIndex: number | undefined;
+}
+
```

Comment:
Since `side` is optional in `NodeLocation`, I believe I can delete `NodeIndex` from types.

Reply:
Though `NodeIndex` contains the possibility of `undefined` values whereas `NodeLocation` doesn't, hmm.